src/Microsoft.AspNet.Mvc.Core/Formatters/JsonPatchInputFormatter.cs

```diff
@@ -86,8 +91,31 @@ public override Task<object> ReadRequestBodyAsync([NotNull] InputFormatterContex
 
                 try
                 {
-                    var contractObject = SerializerSettings.ContractResolver.ResolveContract(type);
-                    return Task.FromResult(jsonSerializer.Deserialize(jsonReader, type));
```

Comment:
I don't understand why this got so much more complicated. Why can't we deserialize the document and then set the property?